internal/services/apimanagement/api_management_policy_fragment_resource.go

```diff
@@ -0,0 +1,232 @@
+	client := meta.(*clients.Client).ApiManagement.PolicyFragmentClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := policyfragment.ParsePolicyFragmentIDInsensitively(d.Get("id").(string))
+	if err != nil {
+		return err
+	}
+
+	format := policyfragment.PolicyFragmentContentFormat(d.Get("format").(string))
+
+	description := d.Get("description").(string)
+	value := d.Get("value").(string)
+
+	parameters := policyfragment.PolicyFragmentContract{
+		Properties: &policyfragment.PolicyFragmentContractProperties{
+			Description: pointer.To(description),
+			Format:      pointer.To(format),
+			Value:       value,
+		},
+	}
+
+	if err := client.CreateOrUpdateThenPoll(ctx, pointer.From(id), parameters, policyfragment.CreateOrUpdateOperationOptions{}); err != nil {
+		return fmt.Errorf("creating/updating %s: %+v", id, err)
+	}
+
+	d.SetId(id.ID())
+
+	return resourceApiManagementPolicyFragmentRead(d, meta)
+}
+
+func resourceApiManagementPolicyFragmentRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).ApiManagement.PolicyFragmentClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := policyfragment.ParsePolicyFragmentIDInsensitively(d.Id())
+	if err != nil {
+		return err
+	}
+
+	format := policyfragment.PolicyFragmentContentFormat(d.Get("format").(string))
+	resp, err := client.Get(ctx, *id, policyfragment.GetOperationOptions{
+		Format: &format,
+	})
+	if err != nil {
+		if response.WasNotFound(resp.HttpResponse) {
+			log.Printf("[DEBUG] %s does not exist - removing from state!", *id)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("retrieving %s: %+v", id, err)
+	}
+
+	d.Set("name", id.PolicyFragmentName)
+	d.Set("resource_group_name", id.ResourceGroupName)
+	d.Set("api_management_name", id.ServiceName)
+
+	if model := resp.Model; model != nil {
+		if props := model.Properties; props != nil {
+			d.Set("description", pointer.From(props.Description))
+			d.Set("value", props.Value)
+			// the api only returns a format field when it's requested in the GET request as param '?format=rawxml' and only does so when it's not 'xml'
+			if props.Format == nil {
+				d.Set("format", "xml")
+			} else {
+				d.Set("format", string(pointer.From(props.Format)))
+			}
+		}
+	}
+
+	return nil
+}
+
+func resourceApiManagementPolicyFragmentDelete(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).ApiManagement.PolicyFragmentClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := policyfragment.ParsePolicyFragmentIDInsensitively(d.Id())
+	if err != nil {
+		return err
+	}
+
+	if resp, err := client.Delete(ctx, *id, policyfragment.DeleteOperationOptions{}); err != nil {
+		if !response.WasNotFound(resp.HttpResponse) {
+			return fmt.Errorf("deleting %s: %s", id, err)
+		}
+	}
```

Comment:
as below can we use `policyfragment.DefaultDeleteOperationOptions` here to handle a default value being set for the options in the future? in addition we don't need to handle the 404 here, so we should be able to reduce this to:

```suggestion
	if err := client.Delete(ctx, *id, policyfragment.DefaultDeleteOperationOptions()); err != nil {
		return fmt.Errorf("deleting %s: %s", *id, err)
	}
```